src/libs/charset.rs

```diff
@@ -1,67 +1,72 @@
-      Charset::Emoji => ['\u{1F000}'..='\u{1F02F}', 
-                         '\u{1F0A0}'..='\u{1F0FF}', 
-                         '\u{1F680}'..='\u{1F6FF}',
-                         '\u{1F910}'..='\u{1F96B}', 
-                         '\u{1F980}'..='\u{1F9E0}'].into(),
+    pub fn get_charset_range(&self) -> Box<[RangeInclusive<char>]> {
+        // non-primitive type return
+        match *self {
+            Charset::AlphaLowercase => ['\u{0061}'..='\u{007A}'].into(),
+            Charset::AlphaUppercase => ['\u{0041}'..='\u{005A}'].into(),
+            Charset::NumSym => ['\u{0021}'..='\u{0040}'].into(),
+            Charset::AlphaNumSym => [
+                '\u{0061}'..='\u{007A}',
+                '\u{0041}'..='\u{005A}',
+                '\u{0021}'..='\u{0040}',
+            ]
+            .into(),
+            Charset::Katakana => ['\u{30A0}'..='\u{30FF}'].into(),
+            Charset::Emoji => [
+                '\u{1F000}'..='\u{1F02F}',
+                '\u{1F0A0}'..='\u{1F0FF}',
+                '\u{1F680}'..='\u{1F6FF}',
+                '\u{1F910}'..='\u{1F96B}',
+                '\u{1F980}'..='\u{1F9E0}',
+            ]
+            .into(),
+        }
     }
-  }
 
-  pub fn get_width(&self) -> CharWidth {
-    match *self {
-      Charset::AlphaLowercase => CharWidth::Half,
-      Charset::AlphaUppercase => CharWidth::Half,
-      Charset::NumSym => CharWidth::Half,
-      Charset::AlphaNumSym => CharWidth::Half,
-      Charset::Katakana => CharWidth::Full, 
-      Charset::Emoji => CharWidth::Full 
-     }
-  }
+    pub fn get_width(&self) -> CharWidth {
+        match *self {
+            Charset::AlphaLowercase => CharWidth::Half,
+            Charset::AlphaUppercase => CharWidth::Half,
+            Charset::NumSym => CharWidth::Half,
+            Charset::AlphaNumSym => CharWidth::Half,
+            Charset::Katakana => CharWidth::Full,
+            Charset::Emoji => CharWidth::Full,
+        }
+    }
 
-  pub fn get_random_char(&self) -> char {
-    let mut rng = rand::thread_rng();
-    let charset = self.get_charset_range();
-    let idx = rng.gen_range(0..charset.len());
-    // Range does not implement iterator over immutable reference..
-    // cloning is cheap for range. just use it
-    rng.gen_range(charset[idx].clone())
-  }
+    pub fn get_random_char(&self) -> char {
+        let mut rng = rand::thread_rng();
+        let charset = self.get_charset_range();
+        let idx = rng.gen_range(0..charset.len());
+        // Range does not implement iterator over immutable reference..
+        // cloning is cheap for range. just use it
+        rng.gen_range(charset[idx].clone())
+    }
 }
 
-
 impl From<CharWidth> for u16 {
     fn from(char_width: CharWidth) -> Self {
         match char_width {
-          CharWidth::Half => 1 as u16,
-          CharWidth::Full => 2 as u16,
+            CharWidth::Half => 1 as u16,
+            CharWidth::Full => 2 as u16,
         }
     }
-}
\ No newline at end of file
+}
```

Comment:
Hey, hello there, thanks for this PR

By the way, does rustfmt work as an extension or is there a config file that can be shared across project?

I've been kinda away from rust since during job I mostly use TS, so pardon me for that.

Reply:
rustfmt is integrated into rust-analyzer and cargo
rust-analyzer should register rustfmt as formatter in whatever editor you use
(for example, in vscode Ctrl+Shift+I should format the current file with rustfmt if rust-analyzer is installed)

you can also run `cargo fmt` on a cargo project to format all rust files in that project